my-api/src/MyList.js

```diff
@@ -0,0 +1,87 @@
+import React, { Component } from "react";
+import MyDetail from "./MyDetail.js";
+import { BrowserRouter as Router, Route, Switch, Link } from "react-router-dom";
+
+export default class MyList extends Component {
+  xx = "aaa";
+  state = {
+    datax: [],
+    loading: true,
+    info: []
+    //xx: "aaa"
+  };
+
+  componentDidMount = () => {
+    fetch("https://rickandmortyapi.com/api/character/")
+      .then(response => response.json())
+      .then(data => {
+        this.setState({ datax: data.results });
+        this.setState({ info: data.info });
+        //console.log({ data });
+      })
+      .catch(error => {
+        this.setState({ error: true });
+        console.log("Oops!");
+      });
+  };
+
+  render() {
+    let myName = 0;
+    let myPic = 0;
+    let myGender = 0;
+    let myStatus = 0;
+    let mySpecies = 0;
+    let myData = this.state.datax.map((x, i) => {
+      //this.xx = x;
+      return (
+        <li key={i}>
+          <img src={x.image} alt="Oops!" className="MyRoundImage" />
+
+          <Link
+            to="/detail"
+            onClick={() => {
+              myName = x.name;
```

Comment:
While this works, it means that refreshing the details page will always cause an "oops" to display. Perhaps after the class ends, look into how you could make another API call from the details page and get the character identifier via a route param.